config.cc

```diff
@@ -28,6 +28,12 @@
 #include <QDesktopServices>
 #endif
 
+#ifdef Q_OS_LINUX
+#define CACHE_DIR_NAME "goldendict"
+#else
+#define CACHE_DIR_NAME "goldendict"
+#endif
```

Comment:
This looks wrong (same string repeated twice). And I think this is not necessary, because filesystem names are case-insensitive on Windows, and macOS's file systems are case-insensitive by default. So no need for this platform-specific code at all.

Reply:
Must have messed this when I was tidying everything. It should be `GoldenDict` in other case. Aren't we concerned here with other OSes?

Reply:
On second thought, probably better not to touch other OSs as "GoldenDict" is more suitable there. So just leave the old code for other systems and use `GenericCacheLocation` only on X11 and Qt 5.

Reply:
I've done that at the beginning, but then I had to repeat `#ifdef Q_OS_LINUX` for Qt5 and Qt4, thus in order to remove repetition, I switched to `GenericCacheLocation`.

Reply:
The problem is gone now that Qt 4 is not touched.